packages/@coorpacademy-components/src/molecule/learning-profile-radar-chart/index.tsx

```diff
@@ -0,0 +1,393 @@
+        dot={
+          <CustomDot
+            onDotClick={handleOnDotClick(datakey)}
+            activeDot={activeDot}
+            dataKey={datakey}
+            stroke={`url(#gradient-stroke-${index})`}
+          />
+        }
+      />
+    );
+  }, totalDataset);
+
+const buildCustomLabel = ({
+  index,
+  x,
+  y,
+  percentagesValues,
+  label,
+  activeDot,
+  chartType,
+  formatedColors
+}: {
+  index: number;
+  x: number;
+  y: number;
+  percentagesValues: number[];
+  label: string;
+  chartType: CHART_TYPE_TYPE;
+  formatedColors: FormatedColorsType[];
+  activeDot?: ActiveDotType;
+}) => {
+  const isCurrentDotActive = activeDot?.label === label;
+  const {
+    offset: {x: offsetX, y: offsetY},
+    alignment,
+    ...rest
+  } = CHART_CONFIGS[chartType].ticks[index];
+
+  return (
+    <g>
+      <foreignObject x={x + offsetX} y={y + offsetY} width="200" height="65">
+        <div
+          className={classnames(style.tickWrapper, isCurrentDotActive && style.tickWrapperFocus)}
+          style={{
+            ...rest,
+            alignItems: alignment,
+            textAlign: alignment,
+            opacity: !isEmpty(activeDot) && !isCurrentDotActive ? 0.3 : 1
+          }}
+        >
+          {formatedColors.map(
+            ({percentage: {color, background}, label: {color: colorLabel}}, i) => (
+              <Fragment key={i}>
+                <span className={style.tickValue} style={{color, background}}>
+                  {percentagesValues[i]}
+                </span>
+                <span className={style.tickLabel} style={{color: colorLabel}}>
+                  {label}
+                </span>
+              </Fragment>
+            )
+          )}
+        </div>
+      </foreignObject>
+    </g>
+  );
+};
+
+/* UTILS */
+const CHART_CONFIGS_BY_SIDE_COUNT = pipe(keyBy('sideCount'), mapValues('name'))(CHART_CONFIGS);
+
+const formatValues: (values_: number | number[]) => Record<string, number> = pipe(
+  values_ => flatten([values_]),
+  values_ => values_.map((val: number, i: number): [string, number] => [`value${i + 1}`, val]),
+  fromPairs
+);
+
+/* this convert incoming component data to rechart data structure */
+export const formatData: (data_: LearningProfileRadarChartPropTypes['data']) => FormatedDataType[] =
+  pipe(
+    toPairs,
+    map(([label, values_]: [string, number | number[]]) => ({
+      ...formatValues(values_),
+      subject: label
+    }))
+  );
+
+const LearningProfileRadarChart = ({
+  data,
+  totalDataset,
+  colors: colorsProps,
+  onClick
+}: LearningProfileRadarChartPropTypes) => {
+  const [isMobile, setIsMobile] = useState(false);
+  const [activeDot, setActiveDot] = useState<ActiveDotType>();
+
+  const formatedData = useMemo(() => formatData(data), [data]);
+
+  const formatedColors = times(i => {
+    if (!colorsProps?.length) return DEFAULT_COLORS;
+
+    const colors = colorsProps[i];
+    return colors ? Object.assign({}, DEFAULT_COLORS, colors) : DEFAULT_COLORS;
```

Comment:
c'est pas équivalent de :
return {...DEFAULT_COLORS, ...colors}

Reply:
(ptetre juste en rajoutant un `as ...`)

Reply:
ici ça va, c'est vraiment pas très couteux, ça fait au maximum 2 itérations,

pour le merge, comme y a typescript, j'évite d'utiliser lodash/fp lorsque y a pas vraiment besoin, 
c'est pas tout le temps le cas mais des fois avec ça pose quelques soucis avec l'inférence des types,
d'ailleurs avant ça, j'avais un spread sur cette ligne mais ça posait problème avec les types